app/controllers/hideout_controller.rb

```diff
@@ -0,0 +1,94 @@
+class HideoutController < ApplicationController
+  def create
+    # Needs owner_email and name as params.
+    # name must be lesser than 20 characters and greater than 3 characters
+    # owner must not already be in a hideout
+    # return the hideout as an object
+    owner_email = params[:owner_email]
+    name = params[:name]
+
+    if !User.exists?(email: owner_email)
+      render status: :not_found, body: "User does not exist"
+    elsif name.length > 20
+      render status: :precondition_failed, body: "Hideout name greater than 20 characters"
+    elsif name.length < 3
+      render status: :precondition_failed, body: "Hideout name lesser than 3 characters"
+    elsif !User.find_by(email: owner_email).hideout_id.nil?
+      render status: :precondition_failed, body: "Owner already in hideout"
+    else
+      hideout_id = Hideout.create(owner_email, name)
+      render status: :created, :json => {name: name, owner_email: owner_email, hideout_id: hideout_id}
+    end
+  end
+
+  def destroy
```

Comment:
Can we just rename this function to delete?

Reply:
rails convention is to call this destroy. By default the route a delete request of the form hideout/:id goes to this method. Could still change this method to delete. But I think its better to follow convention here.

Reply:
Ah ok that's fine then